application-tls-termination/README.md

```diff
@@ -0,0 +1,260 @@
+
+
+## Ingress Controller Configuration
+
+There are Ingress Controller objects created by the Howso Platform Helm chart, but for extended configuration, such as TLS termination at the application, the Ingress Controller objects will often need to be created manually, as Ingress Controllers are often quite different in their configuration.  This K3d example using Traefik will show how to create Ingress rules (using Traefik CRDs) to send encrypted traffic, to the ports of the sidecar containers, for the API, UI, PyPI and UMS services. 
+
+### Some extra background
+
+[Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/) act as a reverse proxy to access services runing in a Kubernetes cluster.  Howso Platform relies on the Ingress controller to route incoming traffic to its service.  Ingress controllers are typically configured by Ingress objects, which layout the traffic routing rules.  The Howso Platform Helm chart will create these objects when it is installed.
+
+> Note: The approach to configuring Ingress Controllers is changing in Kubernetes, with the Ingress object type being frozen, in favor of the [Gateway API](https://kubernetes.io/docs/concepts/services-networking/gateway/).  Support for the Gateway API is increasing, but it is still less common.  Howso Platform does not yet support the Gateway API.
+
+With many implementations of Ingress controller, with a varietey of different features, the Ingress object has not managed to provide a consistent approach outside of the basic routing features.  As a result, many Ingress controllers have their own CRD types, which are more flexible, and can handle more complex routing rules.  It is also common for Ingress Controllers to have a large number of bespoke annotations that can be used against the built in Ingress object.
+
+As a result, the Howso Platform Helm chart is not able to switch on 'Ingress to application' TLS universally for all Ingress controllers.  There are a couple (contour & nginx) that will work just by using the `podTLS.enabled` value, but for most, the Ingress object will need to be disabled, and appropriate Ingress rules created manually.
+
+In this example, we will use Traefik, which comes with the k3d cluster.  It is a common ingress controller in its own right, and serves as a good extended example as configuring it to send TLS traffic to the back end application is not possible with the built in Ingress object.
+
+> Note: Other Ingress controllers will be differ in their configuration, but the [manifests](./manifests) will give a good example of the specific information about paths, ports and service names.
+
+
+## Setup Steps
+
+### Prerequisites
+
+Use the [basic helm install guide](../helm-basic/README.md) to install Howso Platform, and ensure it is running correctly. See the [TLDR](../common/README.md#basic-helm-install) for a quick start.
+
+Install the [step](https://smallstep.com/docs/step-cli/) certificate tool. 
```

Comment:
Are there substantial benefits to using this over the `openssl` CLI tool, which is more standard?

Reply:
Yes (openssl is a pretty horrible cli imho) - but in this case, step cli is used throughout the documentation already, so it sticks with that.